amundsen_application/static/js/config/config-default.ts

```diff
@@ -186,6 +186,12 @@ const configDefault: AppConfig = {
       return `https://DEFAULT_EXPLORE_URL?schema=${schema}&cluster=${cluster}&db=${database}&table=${table}`;
     },
   },
+  tableSource: {
```

Comment:
What do you think about adding an `upstreamSources` configuration to the existing table resource configuration. Let me try to use the types to convey what I mean.

Existing code in config.types:
```
type SourcesConfig = {
  [id: string]: {
    displayName?: string;
    iconClass?: string;
  };
};

interface BaseResourceConfig {
  displayName: string;
  filterCategories?: FilterConfig;
  supportedSources?: SourcesConfig;
}
...
interface ResourceConfig {
  [ResourceType.dashboard]: BaseResourceConfig;
  [ResourceType.table]: BaseResourceConfig;
  [ResourceType.user]: BaseResourceConfig;
}
```

We could extend to:
```
type SourcesConfig = {
  [id: string]: {
    displayName?: string;
    iconClass?: string;
    iconPath?: string; // add new parameter
  };
};

// add new extended interface
interface TableResourceConfig extends BaseResourceConfig {
  upstreamSources?: SourcesConfig;
}

interface ResourceConfig {
  [ResourceType.dashboard]: BaseResourceConfig;
  [ResourceType.table]: TableResourceConfig; // use new interface here
  [ResourceType.user]: BaseResourceConfig;
}
```
Then follow similar patterns in terms of setting up the icons and configuration helper methods (e.g. not importing and accessing `AppConfig` directly).

I think this helps because:
1. It fits nicely into existing precedents by extending them vs. having to create something new that is almost exactly the same.
2. Makes its such this new table related configuration is part of the existing table resource configuration object, and developers won't have to jump around to different spots to configure everything for tables because it will all be part of the same nested object.

What do you think about this?